inc/saiport.h

```diff
@@ -2160,6 +2160,44 @@ typedef enum _sai_port_attr_t
      */
     SAI_PORT_ATTR_MAX_FEC_SYMBOL_ERRORS_DETECTABLE,
 
+    /**
+     * @brief Enable ARS on port. This enables the control for quality measure in switch pipeline.
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_PORT_ATTR_ARS_ENABLE,
```

Comment:
Can a port RIF with disabled ARS be part of the next hop group with ARS enabled? Or is this attribute only related to LAG?

Reply:
Good practice is to enable all the ports for ARS participation and not worry about NHG resolution for a port that is not enabled for ARS.

Having said that if there is a an advantage to disable ARS for certain ports or port groups (like for power etc), and if NHG resolves via port not enabled for ARS, it will effectively not participate in the ARS group and will not get any traffic scheduled. This is easy to observe in a deployment as no traffic will be forwarded to the next hop pointing to this link.